chain/chunks/src/lib.rs

```diff
@@ -995,251 +994,217 @@ impl ShardsManager {
         let (started, key, response) = self.prepare_partial_encoded_chunk_response(request);
 
         let elapsed = started.elapsed().as_secs_f64();
-        let labels = [key, if response.is_some() { "ok" } else { "failed" }];
+        let labels = [
+            key,
+            if response.parts.is_empty() && response.receipts.is_empty() { "failed" } else { "ok" },
+        ];
         metrics::PARTIAL_ENCODED_CHUNK_REQUEST_PROCESSING_TIME
             .with_label_values(&labels)
             .observe(elapsed);
 
-        if let Some(response) = response {
-            self.peer_manager_adapter.do_send(
-                PeerManagerMessageRequest::NetworkRequests(
-                    NetworkRequests::PartialEncodedChunkResponse { route_back, response },
-                )
-                .with_span_context(),
-            );
-            return true;
-        }
-        false
+        self.peer_manager_adapter.do_send(
+            PeerManagerMessageRequest::NetworkRequests(
+                NetworkRequests::PartialEncodedChunkResponse { route_back, response },
+            )
+            .with_span_context(),
+        );
     }
 
     fn prepare_partial_encoded_chunk_response(
         &mut self,
         request: PartialEncodedChunkRequestMsg,
-    ) -> (std::time::Instant, &'static str, Option<PartialEncodedChunkResponseMsg>) {
+    ) -> (std::time::Instant, &'static str, PartialEncodedChunkResponseMsg) {
```

Comment:
Unrelated to this PR, but this started logic looks like it could be simplified. If it always just returns the timestamp, why not move the computation of the timestamp to outside of the function since its calculation has nothing to do with what this function does?

Reply:
Good idea!